jujucrashdump/crashdump.py

```diff
@@ -178,12 +192,29 @@ def __init__(self, model, max_size, extra_dirs, output_dir=None,
         self.timeout = timeout
         self.journalctl = journalctl
 
+    def get_all(self):
+        machines = {}
+        juju_status = yaml.load(open('juju_status.yaml', 'r'),
+                                Loader=yaml.FullLoader)
+        for machine, machine_data in juju_status['machines'].iteritems():
+            machines[machine] = machine_data['ip-addresses']
+            if 'containers' in juju_status['machines'][machine]:
+                for container, container_data in \
+                  juju_status['machines'][machine]['containers'].iteritems():
+                    machines[container] = container_data['ip-addresses']
+        return machines
+
     def _run_all(self, cmd):
-        run_cmd("timeout %ds juju run --all -- sh -c '%s'" % (
-            self.timeout, cmd))
+        all_machines = self.get_all()
+        ssh_cmd = 'ssh -o StrictHostKeyChecking=no'
+        ssh_cmd += ' -i ~/.local/share/juju/ssh/juju_id_rsa'
```

Comment:
Can you build this from somewhere common instead of duplicating the strings twice?

SSH_CMD = ...

Reply:
done